src/librustc_typeck/diagnostics.rs

```diff
@@ -1118,6 +1118,36 @@ fn main() {
 ```
 "##,
 
+E0102: r##"
+You hit this error because the compiler the compiler lacks information
+to determine a type for this variable. Erroneous code example:
+
+```
+fn main() {
+    let x: &_; // error: cannot determine a type for this local variable
+}
+```
+
+You have two possibilities to solve this situation:
+ * Give an explicit definition of the variable
+ * Infer the variable
```

Comment:
Then:

“Help the compiler to infer the variable” or somehing.

---

Basically to infer a variable an explicit type definition has to occur somewhere anyway (even if it is in a function parameter list; notable exception being integral variables which have a default of i32), so I don’t see much reason to add this alternative.